src/ui/zabapgit_js_common.w3mi.data.js

```diff
@@ -1205,67 +1146,58 @@ Patch.prototype.registerClickHandlerForSelector = function(sSelector, fnCallback
 
 };
 
-Patch.prototype.registerClickHandlerForPatchFile = function(sSelector){
-  this.registerClickHandlerForSelector(sSelector, this.patchLinkClickFile);
-};
-
-Patch.prototype.registerClickHandlerForPatchSection = function(sSelector){
-  this.registerClickHandlerForSelector(sSelector, this.patchLinkClickSection);
-};
-
-Patch.prototype.registerClickHandlerForPatchLine = function(sSelector) {
-  this.registerClickHandlerForSelector(sSelector, this.patchLinkClickLine);
-};
-
-Patch.prototype.patchLinkClickLine = function(oEvent){
-  this.togglePatchForElem(oEvent.srcElement);
-  oEvent.preventDefault();
+Patch.prototype.getAllLineCheckboxesForId = function(sId, sIdPrefix){
+  var oRegex = new RegExp("^" + sIdPrefix);
+  sId = sId.replace(oRegex, PatchLine.prototype.ID);
+  return document.querySelectorAll("input[id^='"+ this.escape(sId) + "']");
 };
 
-Patch.prototype.togglePatchForElem = function(elLink) {
-  new PatchLine(elLink.id).toggle();
+Patch.prototype.getAllSectionCheckboxesForId = function(sId, sIdPrefix){
+  var oRegex = new RegExp("^" + sIdPrefix);
+  sId = sId.replace(oRegex, PatchSection.prototype.ID);
+  return document.querySelectorAll("input[id^='"+ this.escape(sId) + "']");

```

Comment:
Since the files are patched one by one, you could narrow the query to the section with the patched file instead of the whole document to boost performance even further.

Reply:
Also, getElementById can be used here, which is faster than the general querySelectorAll
https://web.archive.org/web/20160108040024/http://jsperf.com/getelementbyid-vs-queryselector

Reply:
Hm, I'm not quite sure how to do that. General idea is that you have the section id (e.g. patch_section_zbunkai_christian.prog.abap_1) and then find all checkboxes starting with the same id prefix (e.g. patch_**line**_zbunkai_christian.prog.abap_1).

![image](https://user-images.githubusercontent.com/17437789/57987482-409ed900-7a82-11e9-89c6-a22d2fadb757.png)

Imho the query is quite narrow and getElementById cannot be used, because we don't know the exact id and we'd like to select multiple elements. Or am I missing the obvious?

Btw. perfomance improved already dramatically. Patching objects with 5000+ changed lines of code withinin the blink of an eye.

Reply:
My bad, I thought you were retrieving each line by its id separately.  The checkboxes should be more than enough then.